libraries/chain/controller.cpp

```diff
@@ -691,10 +700,19 @@ struct building_block {
                   .new_protocol_feature_activations = new_protocol_feature_activations()
                };
 
+               // Get parent block reference.
+               auto parent_timestamp = timestamp();
+               parent_timestamp.slot--;
```

Comment:
The parent timestamp is not necessarily the previous time slot. Instead I think this should be `chain_head->timestamp()`

Reply:
Thanks. I should have used `bb.parent.timestamp()`; it was too late last night to go through the code.